### PR TITLE
log: Allow for disable stack traces from log.Fatal in tests 

### LINKED ISSUE #16128: migrations: TestLeaseExpiration prints a stack trace on success
Because this test induces a call to `log.Fatal`, it produces this output in CI:
```
...
=== RUN   TestLeaseExpiration
F170522 22:23:41.914073 58 migrations/migrations.go:244  not enough time left on migration lease, terminating for safety
goroutine 58 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x33e4b01, 0x367ba947, 0x2a23ca0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:886 +0xc4
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2a24400, 0xc400000004, 0x2584adb, 0x18, 0xf4, 0xc42020a180, 0x3f)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:754 +0x6e7
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x7f36c12bd1e0, 0xc420074d78, 0xc400000004, 0x2, 0x0, 0x0, 0xc42003dea8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:158 +0x5ea
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x7f36c12bd1e0, 0xc420074d78, 0x1, 0x4, 0x0, 0x0, 0xc42003dea8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:68 +0x9a
github.com/cockroachdb/cockroach/pkg/util/log.Fatal(0x7f36c12bd1e0, 0xc420074d78, 0xc42003dea8, 0x1, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:171 +0x88
github.com/cockroachdb/cockroach/pkg/migrations.(*Manager).EnsureMigrations.func2(0x7f36c12bd1e0, 0xc420074d78)
	/go/src/github.com/cockroachdb/cockroach/pkg/migrations/migrations.go:244 +0x3c6
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc4200a68c0, 0x7f36c12bd1e0, 0xc420074d78, 0x2584adb, 0x18, 0xf8, 0x0, 0x0, 0xc4202dc480)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:294 +0x101
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:295 +0x173

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4203dc410, 0x1c8d3b2, 0x13, 0x1d244f8, 0xc420677c01)
	/usr/local/go/src/testing/testing.go:698 +0x582
testing.runTests.func1(0xc4203dc410)
	/usr/local/go/src/testing/testing.go:882 +0xab
testing.tRunner(0xc4203dc410, 0xc420677d78)
	/usr/local/go/src/testing/testing.go:657 +0x108
testing.runTests(0xc4202dc320, 0x2832fe0, 0x4, 0x4, 0x0)
	/usr/local/go/src/testing/testing.go:888 +0x4e1
testing.(*M).Run(0xc420cc3f28, 0xc420677ed0)
	/usr/local/go/src/testing/testing.go:822 +0x1c4
main.main()
	github.com/cockroachdb/cockroach/pkg/migrations/_test/_testmain.go:48 +0x210

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2197 +0x1

goroutine 19 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x2a24400)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1052 +0x85
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:594 +0x13a

goroutine 20 [syscall]:
os/signal.signal_recv(0x6d0131)
	/usr/local/go/src/runtime/sigqueue.go:116 +0x104
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 57 [select]:
github.com/cockroachdb/cockroach/pkg/migrations.TestLeaseExpiration.func3(0x7f36c12bd1e0, 0xc420074d78, 0x2850aa0, 0xc4203b80c0, 0x0, 0x0, 0x1, 0x1a)
	/go/src/github.com/cockroachdb/cockroach/pkg/migrations/migrations_test.go:407 +0x187
github.com/cockroachdb/cockroach/pkg/migrations.(*Manager).EnsureMigrations(0xc4201b3a40, 0x7f36c12bd1e0, 0xc420074d78, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/migrations/migrations.go:274 +0xda9
github.com/cockroachdb/cockroach/pkg/migrations.TestLeaseExpiration(0xc42007c0d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/migrations/migrations_test.go:416 +0x66e
testing.tRunner(0xc42007c0d0, 0x1d244f8)
	/usr/local/go/src/testing/testing.go:657 +0x108
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:697 +0x544
--- PASS: TestLeaseExpiration (0.01s)
PASS
ok  	github.com/cockroachdb/cockroach/pkg/migrations	1.374s
```

Since this test already mocks out the log package's `exitFunc`, it'd be nice to also suppress stack printing.

### PR DESCRIPTION
**Pull request by a-robinson**
Also remove an unused variable while I'm here.

Fixes #16128

### CODE REVIEW

File: pkg/migrations/migrations_test.go [hunk 1]
```diff
@@ -400,6 +400,8 @@ func TestLeaseExpiration(t *testing.T) {
 	exitCalled := make(chan bool)
 	log.SetExitFunc(func(int) { exitCalled <- true })
 	defer log.SetExitFunc(os.Exit)
+	// Disable stack traces to make the test output in teamcity less deceiving.
+	defer log.DisableTracebacks()()
```
Comment (tschottdorf) [inline pkg/migrations/migrations_test.go]: Looks racy, WDYT?

**Comment by tamird** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16135#-:-Kl-QVFVR05-Du4msw4d:bnfp4nl)*

File: pkg/migrations/migrations_test.go (hunk 1 above)
Comment (a-robinson) [inline pkg/migrations/migrations_test.go]: Potentially, yeah. Although as long as `DisableTracebacks` is only used in tests, it's not going to break anything. Do you have a different suggestion?

Comment (tschottdorf) [inline pkg/migrations/migrations_test.go]: If it's fine by @tamird, it's fine by me.

**Pull request merged**